deeprankcore/dataset.py

```diff
@@ -658,17 +663,20 @@ def __init__( # pylint: disable=too-many-arguments, too-many-locals
         if not train:
             if not isinstance(dataset_train, GraphDataset):
                 raise TypeError("Please provide a valid training GraphDataset.")
-
+            
+            #check inherited parameter with the ones in the training set
+            inherited_param = ["target", "task", "node_features", "edge_features", "features_dict", "classes", "target_transform", "features_transform"]
+            self._check_inherited_param(inherited_param, dataset_train)
+            
+            #set features_transform as the same in dataset_train
```

Comment:
Line 672 and below now are not needed anymore (that check is in `_check_inherited_param`)